### DISCUSSION TITLE
Dropout Not Working During Model Training

### DISCUSSION THREAD

**Discussion by Lingpheng** [ultralytics/ultralytics#20902] [Q&A]
Hi Ultralytics team,

I tried to add dropout when training my model, but it doesn’t seem to have any effect. Here is how I set it:

```
results = model.train(
    data=data_path,
    epochs=20,
    patience=15,
    batch=32,
    imgsz=640,
    device='0,1',
    seed=88,
    workers=cpu_cores,
    pretrained=False,
    optimizer='AdamW',
    lr0=0.001,
    lrf=0.1,
    dropout=0.3,
    project="runs",
    save_period=40,
    cls=0.8,
    close_mosaic=0,
)
```
To check whether the dropout parameter works, I also tried training with dropout=0:

```
results = model.train(
    data=data_path,
    epochs=20,
    patience=15,
    batch=32,
    imgsz=640,
    device='0,1',
    seed=88,
    workers=cpu_cores,
    pretrained=False,
    optimizer='AdamW',
    lr0=0.001,
    lrf=0.1,
    dropout=0,
    project="runs",
    save_period=40,
    cls=0.8,
    close_mosaic=0,
)
```

However, the resulting models from both runs are exactly the same.

Here's the comparison image showing identical performance:

![Dropout](https://github.com/user-attachments/assets/cb68047b-f839-4629-a493-be839de872a4)

Why isn't the dropout parameter having any effect?

Thanks in advance for your help!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Lingpheng, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. 

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response. An Ultralytics engineer will also assist you here soon 😊.

**Reply by glenn-jocher** [reply]
Hi @Lingpheng! The `dropout` parameter in YOLO training settings is specifically designed for [classification tasks](https://docs.ultralytics.com/modes/train/#train-settings), not detection tasks. Since you appear to be training a detection model (evidenced by parameters like `cls=0.8` for classification loss weight), the dropout parameter has no effect on your model architecture. For detection models, YOLO uses other regularization techniques built into the architecture, which is why you're seeing identical results regardless of the dropout value.

**Reply by Lingpheng** [reply]
Hi @glenn-jocher, that makes perfect sense! So there's no need to manually add a dropout parameter for detection tasks, since YOLO already handles it internally. Thanks a lot for the explication.

**Reply by glenn-jocher** [reply]
You're absolutely right! YOLO detection models use other effective regularization techniques like weight decay, data augmentation (mosaic, mixup), and architectural designs that inherently prevent overfitting without needing explicit dropout layers. Glad that cleared things up for you! 😊
